app/Services/DomainService.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Services;
+
+use Carbon\Carbon;
+use Iodev\Whois\Factory;
+use App\Models\Monitor;
+use App\Notifications\Notifiable;
+use App\Notifications\DomainExpiration;
+
+class DomainService
+{
+    protected $whois;
+
+    public function __construct()
+    {
+        $this->whois = Factory::get()->createWhois();
+    }
+
+    public function lookupDomain($url)
+    {
+        $baseDomain = $this->getBaseDomainFromUrl($url);
+
+        $domainInfo = $this->whois->loadDomainInfo($baseDomain);
+
+        if ($domainInfo && $domainInfo->expirationDate) {
+            return response()->json([
+                'expiration_date' => date('Y-m-d H:i:s', $domainInfo->expirationDate)
+            ]);
+        }
+        return 0;
```

Comment:
I used these return values to determine if the domain details were being retrieved or not. 
But yes, I can use boolean values instead of integer values for it.